### ISSUE TITLE
Connections: createSegment() method has missing break for short circuiting

### ISSUE THREAD

**Issue by cogmission** [numenta/nupic#3284]
Found a slight (but important) oversight.
[This for loop](https://github.com/numenta/nupic/blob/master/src/nupic/research/connections.py#L323) should have a ```break``` statement on line 327. This will avoid looping through all of the segments when after a Segment marked "destroyed" is already found.

The correct behavior is executed for the [createSynapse() method here](https://github.com/numenta/nupic/blob/master/src/nupic/research/connections.py#L398) where you will see a ```break``` statement is used.

**Comment by rhyolight** [comment]
@scottpurdy or @mrcslws can you take a quick look to validate this concern?

**Comment by mrcslws** [comment]
You're correct, this could be better. The issue doesn't currently affect TemporalMemory, since there is never more than one destroyed segment. But it's pretty easy to exhibit the issue with a few calls to `destroySegment` and `createSegment` (e.g. in a Connections unit test).

In general this won't cause any "wrong" behavior, it will just be inefficient. I agree with changing it. The issue exists in the [C++ code](https://github.com/numenta/nupic.core/blob/4063c8cc0fa0dcf70c72481a5d839738cebc9868/src/nupic/algorithms/Connections.cpp#L94) as well.
